packages/swapper/src/swappers/ThorchainSwapper/endpoints.ts

```diff
@@ -360,45 +359,46 @@ export const thorchainApi: SwapperApi = {
       const thorTxHash = txHash.replace(/^0x/, '')
 
       // not using monadic axios, this is intentional for simplicity in this non-monadic context
-      const { data } = await axios.get<ThornodeStatusResponse>(
-        `${config.REACT_APP_THORCHAIN_NODE_URL}/lcd/thorchain/tx/status/${thorTxHash}`,
-      )
-
-      if ('error' in data) {
+      const [{ data: txData }, { data: txStatusData }] = await Promise.all([
+        axios.get<ThornodeTxResponse>(
+          `${config.REACT_APP_THORCHAIN_NODE_URL}/lcd/thorchain/tx/${thorTxHash}`,
+        ),
+        axios.get<ThornodeStatusResponse>(
+          `${config.REACT_APP_THORCHAIN_NODE_URL}/lcd/thorchain/tx/status/${thorTxHash}`,
+        ),
+      ])
+
+      // We care about txStatusData errors because it drives all of the status logic.
+      if ('error' in txStatusData) {
         return {
           buyTxHash: undefined,
           status: TxStatus.Unknown,
           message: undefined,
         }
       }
 
-      const latestOutTx = data.out_txs?.[data.out_txs.length - 1]
-      const hasOutboundTx = latestOutTx?.chain !== 'THOR'
+      // We use planned_out_txs to determine the number of out txs because we don't want to derive
+      // swap completion based on the length of out_txs which is populated as the trade executed
+      const numOutTxs = txStatusData.planned_out_txs?.length ?? 0
+      const lastOutTx = txStatusData.out_txs?.[numOutTxs - 1]
 
-      const buyTxHash = parseThorBuyTxHash(txHash, latestOutTx)
+      const buyTxHash = parseThorBuyTxHash(txHash, lastOutTx)
 
-      // if we have an outbound transaction (non rune) and associated buyTxHash, check if it's been confirmed on-chain
-      if (hasOutboundTx && buyTxHash) {
-        const outboundTxConfirmations = await checkOutboundTxConfirmations(
+      // If thornode says the tx is done, its done ;)
+      // We discard txData errors because we can derive the status before it returns a success response.
```

Comment:
q: Not sure if I got this comment right, but do we explicitly *not* want to early return on `error` in `txData` the same way we do for `txStatusData`? Assuming this would mostly be a paranoia check because if there is an error in the Tx, it would most likely be caught in the status error checks above, but would make things some cleaner.

Reply:
Per comment below, yeah we want to ignore error state of the txData response. We cannot remove the `!('error' in txData)` because it's what allows TS to determine the available fields on the success response type, so if we did remove it we'd have nullish checks everywhere below this line with no functional difference